javascripts/2048.js

```diff
@@ -1,32 +1,276 @@
 $(document).ready(function() {
   console.log('ready!');
+  arrayInitialize();
+  initializeGame();
   $('body').keydown(function(event){
     var arrow_keys = [37, 38, 39, 40];
+
     if(arrow_keys.indexOf(event.which) > -1) {
-      var tile = $('.tile');
-      moveTile(tile, event.which);
+      // var tile = $('.tile');
+      moveTile(event.which);
       event.preventDefault();
     }
   })
 })
 
-function moveTile(tile, direction) {
+function pop(tile) {
+  $(tile).addClass("popper")
+  .on("animationend",function() {
+    $(this).removeClass("popper");
+  });
+}
+
+
+ //keeps track of where on the board has a tile
+var tile_array = [];
```

Comment:
When using global vars like this, it's useful to put all those at the top of the file so they're easy to reference.